lib/blockscore/collection.rb

```diff
@@ -1,18 +1,91 @@
 module BlockScore
   class Collection < Array
-    def initialize(target)
-      super()
-      @target = target
+    attr_reader :default_params
+    protected :default_params
+
+    attr_reader :parent, :target, :data
+
+    def initialize(params)
+      @parent = params.fetch :parent
+      @target = params.fetch :target
+      register_parent_data
+    end
+
+    def default_params
+      {
+        :"#{parent_name}_id" => parent_id
+      }
+    end
+
+    def parent_id
+      parent.id if parent.respond_to?(:id)
+    end
+
+    def all
+      self
+    end
+
+    def new(params = {})
+      item = target.new(params.merge(default_params))
+      ctxt = self
+      item.define_singleton_method(:save) do
+        ctxt.parent.save unless ctxt.parent_id
+        send :"#{ctxt.parent_name}_id=", ctxt.parent_id
+        super()
+      end
+      self << item
+      item
+    end
```

Comment:
when removed I believe the QuestionSet instance tries to call `parent` on itself